src/app/[...slug]/page.tsx

```diff
@@ -73,26 +71,31 @@ export default async function Page({ params, searchParams }: PageParams) {
     }
 
     const layout = layoutOrError as PageLayoutServiceResponse;
+    const isEdit = searchParams['sfaction'] === 'edit';

```

Comment:
`preview` and `edit` actions could be extracted to a common place, for example `enum Action` and get from there. Not sure if you will need further to check for actions, however it will be not hardcoded as a string here

Reply:
It will be cool - It is a common issue that we discussed earlier to tackle in a separate pr as it related to more widgets so I am leaving it like this for the moment.